docs/configuration/index.md

```diff
@@ -24,27 +24,94 @@ The configuration settings are read from two sources:
+- upgrade BotKube installation with Helm,
+- or use dedicated `@BotKube` commands, to e.g. toggle notifications or edit Source Bindings. See the [Usage](../usage/index.md) section for more details.
+
+If you wish to change the configuration with Helm, create a `/tmp/values.yaml` file that contains the new values and use the **helm upgrade** command:
 
 ```bash
-# Change resources related settings
-kubectl edit configmap botkube-global-config -n botkube
+helm upgrade -n botkube botkube -f /tmp/values.yaml helm/botkube --wait
 ```
 
-```bash
-# Change communication related settings
-kubectl edit secret botkube-communication-secret -n botkube
+As both Helm release upgrade and some of the `@BotKube` commands modify the same configuration, it is merged during the **helm upgrade** command.
+Whenever you specify a new value in the `/tmp/values.yaml` file, it will override the existing value in the configuration.
+
+### Preventing overrides by default Helm chart values
+
+Keep in mind that even if you don't specify custom values in the `/tmp/values.yaml` file, Helm can override the existing values with the default ones.
+
+Consider the following config:
+
+```yaml
+communications:
+  "default-group":
+    socketSlack:
+      enabled: true
+      botToken: "{botToken}"
+      appToken: "{appToken}"
+      channels:
+        "default":
+          name: general
+          notification:
+            disabled: false # default from the Helm chart
+          bindings:
+            sources:
+              - k8s-all-events # default from the Helm chart
+# (...)
 ```
 
-This command opens ConfigMap `specs` in default editor. Do the required changes, save and exit. The BotKube Pod will automatically restart to have these configurations in effect.
+Assume that users ran the following commands:
 
-## Helm install options
+```
+@BotKube edit SourceBindings k8s-err-events, k8s-recommendation-events
+@BotKube notifier stop
+```
 
-Advanced Helm install options are documented [here](helm-chart-parameters).
+Which effectively result in the following config that BotKube sees:
+
+```yaml
+communications:
+  "default-group":
+    socketSlack:
+      enabled: true
+      botToken: "{botToken}"
+      appToken: "{appToken}"
+      channels:
+        "default":
+          name: general
+          notification:
+            disabled: true # set by user command
+          bindings:
+            sources:
+              - k8s-err-events # set by user command
+              - k8s-recommendation-events # set by user command
+# (...)
+```
+
+To persist the configuration that users provided, and not overwrite notification and source bindings values, run Helm upgrade with:
+
+```yaml
+communications:
+  "default-group":
+    socketSlack:
+      channels:
+        "default":
+          name: general
+          notification: null # explicitly not use defaults from Helm chart
+          bindings:
+            sources: null # explicitly not use defaults from Helm chart
```

Comment:
can I do that for any property? or there are some rules? If yes, it would be nice to have a list of them e.g.:

- `communications.{name}.socketSlack.channels.{name}.notifications`
- `communications.{name}.socketSlack.channels.{name}.bindings.sources`
- etc.
---
(please ignore this comment if not valid, I'm not aware enough about current merge policy)

Reply:
Good point, added such info 👍